docs/android/ci.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Android Continuous Integration and Delivery"
+sidebar_label: "Continuous Integration"
+---
+
+🚧🚧🚧 Under construction 🚧🚧🚧
+
+## Android Continuous Integration and Delivery
+
+This document outlines the Continuous Integration (CI) and Continuous Delivery (CD) processes for the Android project. We use **GitHub Actions** as our CI/CD platform, with multiple workflows configured to ensure code quality, automate builds, and streamline deployments.
+
+## Overview
+
+The main goals of our CI/CD process are:
+
+- ✅ Validate that everything is working as expected.
+- 🚨 Notify relevant people if something breaks.
+- 🚀 Enable fully automated continuous delivery of applications.
+- 🔄 Avoid duplication by extracting common code into reusable local actions under `.github/actions`.
+
+## Versioning
+
+We follow the same versioning convention as the core project, using [CalVer] (Calendar Versioning). This ensures consistency across all releases.
+
+## Workflows
+
+### On Pull Request
+
+When a pull request (PR) is opened or updated, the `pr.yml` workflow is triggered. Its goals are:
+
+- 🧹 Validate code compliance with our [linters](linter).
+- 🔨 Ensure the code builds successfully.
+- ✅ Run all tests to verify correctness.
+- 📦 Persist generated APKs in the GitHub Actions tab for review.
+
+If any step fails:
+
+- The CI notifies the PR owner.
+- The PR is blocked from being merged until the issues are resolved.
+- Fixes must be committed, which automatically restarts the workflow.
+
+:::note
+Only one workflow runs at a time for a given PR. If multiple commits are pushed in quick succession, the CI cancels ongoing builds and processes only the latest commit.
+:::
+
+#### Debug Builds
+
+To build the application in debug on CI, we use a mock Google services file located at `/.github/mock-google-services.json`.
+
+### On Push to `main`
+
+When a commit is pushed to the `main` branch, the `onPush.yml` workflow is triggered. Its goals are:
+
+- 🌐 Download translations from [Lokalise](../translations).
+- 📝 Generate release notes.
+- 🔧 Build release variants of all applications.
+- 📤 Deploy applications to Firebase.
+- 🛒 Deploy to the internal track of the Play Store.
+- 📦 Persist generated APKs in the GitHub Actions tab.
+- 🔐 Inject secrets and files required for publishing.
+
+We use [Fastlane](https://fastlane.tools/) to simplify deployment to different stores. All Fastlane configurations can be found in the `fastlane` folder.
+
+:::note
+This workflow can also be manually triggered with the `beta` flag to promote a build to the beta track on the stores.
```

Comment:
Technically true, but if you want a new beta you usually trigger the weekly builds workflow so it also creates a GitHub prerelase.

Reply:
I speak about it a bit bellow but I can remove this note if you prefer.